bfvmm/src/vmcs/src/vmcs_intel_x64.cpp

```diff
@@ -298,36 +285,51 @@ vmcs_intel_x64::save_state()
 }
 
 vmcs_error::type
+vmcs_intel_x64::unlaunch()
+{
+    m_intrinsics->write_es(m_intrinsics->vmread(VMCS_GUEST_ES_SELECTOR));
+    m_intrinsics->write_ds(m_intrinsics->vmread(VMCS_GUEST_DS_SELECTOR));
+    m_intrinsics->write_fs(m_intrinsics->vmread(VMCS_GUEST_FS_SELECTOR));
+    m_intrinsics->write_gs(m_intrinsics->vmread(VMCS_GUEST_GS_SELECTOR));
+    m_intrinsics->write_msr(IA32_EFER_MSR, m_intrinsics->vmread(VMCS_GUEST_IA32_EFER_FULL));
+    m_intrinsics->write_msr(IA32_PAT_MSR, m_intrinsics->vmread(VMCS_GUEST_IA32_PAT_FULL));
+    m_intrinsics->write_msr(IA32_SYSENTER_CS_MSR, m_intrinsics->vmread(VMCS_GUEST_IA32_SYSENTER_CS));
+    m_intrinsics->write_msr(IA32_FS_BASE_MSR, m_intrinsics->vmread(VMCS_GUEST_FS_BASE));
+    m_intrinsics->write_msr(IA32_GS_BASE_MSR, m_intrinsics->vmread(VMCS_GUEST_GS_BASE));
+    m_intrinsics->write_msr(IA32_SYSENTER_ESP_MSR, m_intrinsics->vmread(VMCS_GUEST_IA32_SYSENTER_ESP));
+    m_intrinsics->write_msr(IA32_SYSENTER_EIP_MSR, m_intrinsics->vmread(VMCS_GUEST_IA32_SYSENTER_EIP));
+    m_intrinsics->write_cr3(m_intrinsics->vmread(VMCS_GUEST_CR3));
+
+    promote_vmcs_to_root();
+
```

Comment:
I would get rid of the extra space and comment. Instead, consider marking as noreturn